components/brave_wallet_ui/utils/style.utils.ts

```diff
@@ -32,60 +32,64 @@ export const getDominantColorFromImageURL = (src: string) => {
     return undefined
   }
 
-  // Construct new image from src.
-  let img = new Image()
-  img.src = src
-
-  // Images actual width and height.
-  const { naturalWidth, naturalHeight } = img
-
-  // Higher number will have better precision but may be slower.
-  const precision = 10
-
-  // Calculate canvas sizes.
-  const scaleRatio = Math.min(
-    precision / naturalWidth,
-    precision / naturalHeight
-  )
-  const width = Math.ceil(naturalWidth * scaleRatio)
-  const height = Math.ceil(naturalHeight * scaleRatio)
-  const area = width * height
-
-  // area might be 0.
-  if (!area) {
+  try {
+    // Construct new image from src.
+    let img = new Image()
+    img.src = src
+
+    // Images actual width and height.
+    const { naturalWidth, naturalHeight } = img
+
+    // Higher number will have better precision but may be slower.
+    const precision = 10
+
+    // Calculate canvas sizes.
+    const scaleRatio = Math.min(
+      precision / naturalWidth,
+      precision / naturalHeight
+    )
+    const width = Math.ceil(naturalWidth * scaleRatio)
+    const height = Math.ceil(naturalHeight * scaleRatio)
+    const area = width * height
+
+    // area might be 0.
+    if (!area) {
+      return undefined
+    }
+    // Construct canvas.
+    let context = document.createElement('canvas').getContext('2d')
+
+    // context may be null.
+    if (!context) {
+      return undefined
+    }
+
+    // Set canvase width and height.
+    context.canvas.width = width
+    context.canvas.height = height
+
+    // Draw image to canvas.
+    context.drawImage(img, 0, 0, width, height)
+
+    // Get raw image data from canvas.
+    const data = context.getImageData(0, 0, width, height).data
+    let r = 0
+    let g = 0
+    let b = 0
+
+    // Compare and find average color values
+    for (let i = 0; i < data.length; i += 4) {
+      r += data[i]
+      g += data[i + 1]
+      b += data[i + 2]
+    }
+
+    r = ~~(r / area)
+    g = ~~(g / area)
+    b = ~~(b / area)
+
+    return `rgba(${r},${g},${b},0.2)`
+  } catch {
     return undefined
   }
-  // Construct canvas.
-  let context = document.createElement('canvas').getContext('2d')
-
-  // context may be null.
-  if (!context) {
-    return undefined
-  }
-
-  // Set canvase width and height.
-  context.canvas.width = width
-  context.canvas.height = height
-
-  // Draw image to canvas.
-  context.drawImage(img, 0, 0, width, height)
-
-  // Get raw image data from canvas.
-  const data = context.getImageData(0, 0, width, height).data
-  let r = 0
-  let g = 0
-  let b = 0
-
-  // Compare and find average color values
-  for (let i = 0; i < data.length; i += 4) {
-    r += data[i]
-    g += data[i + 1]
-    b += data[i + 2]
-  }
-
-  r = ~~(r / area)
-  g = ~~(g / area)
-  b = ~~(b / area)
-
-  return `rgba(${r},${g},${b},0.2)`
 }
```

Comment:
why we need this?

Reply:
WebKit can fail for images from different domains with `Unable to get image data from canvas because the canvas has been tainted by cross-origin data.`, causing full page blank render. With try/catch we return `undefined` instead and will just fail to fetch dominant colour rather than blanking the entire page render.